### ISSUE TITLE
Polygon distortions on high zoom levels

### ISSUE THREAD

**Issue by lukasvdelft** [mapbox/mapbox-gl-js#12825]
**mapbox-gl-js version**: 2.15.0
**mapbox-gl-draw**: 1.4.0

**browser**: Chrome version 114.0.5735.198

### Actual Behavior

<img width="947" alt="Screenshot 2023-08-02 at 12 54 50" src="https://github.com/mapbox/mapbox-gl-js/assets/103726718/bd6f563a-fa1d-4113-ad3e-07e4c157f41b">

![Screenshot 2023-08-02 at 12 54 39](https://github.com/mapbox/mapbox-gl-js/assets/103726718/68ebdda6-9e3e-4bde-9b57-b3d218c0eba5)

![Screenshot 2023-08-02 at 12 49 11](https://github.com/mapbox/mapbox-gl-js/assets/103726718/f60daab7-267b-473f-9a07-3721685cd47b)

**Comment by mourner** [comment]
In general you can increase precision for GeoJSON sources in GL JS by setting e.g. `maxzoom: 24` in the source options. However, since the issue is about features coming from `mapbox-gl-draw` specifically, this might require implementing a `maxzoom` option in the plugin itself, if I understood your case correctly. Can you show a sample live test case?

**Comment by lukasvdelft** [comment]
yes that is exactly what I was expecting, that the maxzoom of gl-draw is the limiting (distorting) factor. i am feeding GeoJson to the draw.add function. in the issue list of gl-draw someone requested a higher zoom factor as well (#1143), but that was in januari, so it is probably not something on the radar. would it matter if I requested that feature as well? for architecture use-cases, the precision as is is surely not sufficient. I don't have a live case yet as I prioritised other things first. I would prefer starting development with mapbox-gl. thanks so far.

**Comment by mourner** [comment]
@lukasvdelft would you like to submit a PR to GL Draw? The actual change should be pretty small and easy.

**Comment by lukasvdelft** [comment]
I have the impression that the maxzoom is hardcoded on the GL side (the last PR on this I found is yours, https://github.com/mapbox/mapbox-gl-js/commit/d6c34c81f7b0d6e77f1b25c2c080a3c5afba94ab).

Draw loads layers with "style" type geojson 

ctx.options.styles.forEach(function (style) {
        ctx.map.addLayer(style);
      }); 

So how could Draw introduce a a maxzoom for geojson if it is limited on GL side? I probably don't understand enough of the whole setup to judge this correctly.

**Comment by lukasvdelft** [comment]
ah, I see in setup.js of mapbox-gl-draw a function addLayers(). it specifies a sources.COLD and sources.HOT, like: 

ctx.map.addSource(sources.COLD, {
        data: {
          type: geojsonTypes.FEATURE_COLLECTION,
          features: []
        },
        type: 'geojson'
}); 

does that need to be extended with a variable which sets maxzoom? like:

ctx.map.addSource(sources.COLD, {
        data: {
          type: geojsonTypes.FEATURE_COLLECTION,
          features: []
        },
        type: 'geojson',
        maxzoom: defaultOptions.maxZoom
}); 

and add maxZoom to the const defaultOptions

I never made a PR before
